app/assets/javascripts/filters.js

```diff
@@ -10,7 +10,9 @@ $(document).ready(function () {
   });
 
   $('#filter_unlimited').change(function () {
-    $('#search').prop('disabled', $(this).prop('checked'));
+    var isChecked = $(this).prop('checked');
+    var autocompleteID = $('.search-input').data('autocompleteId');
+    tfm.autocomplete.onUnlimitedChange(isChecked, autocompleteID);
```

Comment:
`onUnlimitedChange` feel confusing here because I know when it should happen (on-unlimited-change) but I have no idea what it actually going to do.
I see it important because we might want to reuse the disable/enable in different scenarios.

What do you think about changing to:
```js
if (isChecked) {
  tfm.autocomplete.disableAutocomplete(autocompleteID);
} else {
  tfm.autocomplete.enableAutocomplete(autocompleteID);
}
```